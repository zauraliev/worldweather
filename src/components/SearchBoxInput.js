import React from "react";

const SearchBoxInput = props => (
  <div className="form-group">
    <input
      type="text"
      name={props.inputName}
      className="form-control"
      id={props.inputId}
      placeholder={props.placeholder}
      onChange={props.updateFormInputs}
      onBlur={props.updateFormInputs}
      required={props.errorType}
    />
    {props.errorType ? (
      <div className="invalid-feedback">Please enter {props.placeholder} field</div>
    ) : null}
  </div>
);

export default SearchBoxInput;
