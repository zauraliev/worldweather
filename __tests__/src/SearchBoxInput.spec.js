import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SearchBoxInput from "../../src/components/SearchBoxInput";

describe("SearchBoxInput", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchBoxInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
