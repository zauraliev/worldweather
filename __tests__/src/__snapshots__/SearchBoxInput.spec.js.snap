// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxInput renders correctly 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    type="text"
  />
</div>
`;
